packages/dev/core/src/Shaders/pbr.fragment.fx

```diff
@@ -678,6 +678,26 @@ void main(void) {
             gl_FragData[PREPASS_REFLECTIVITY_INDEX] = vec4(0.0, 0.0, 0.0, writeGeometryInfo);
         #endif
     #endif
+        #ifdef PREPASS_SPECULARGLOSSINESS

```

Comment:
I am really not sure about the maths in all this part.

I guess you should rely on roughness and specularEnvironmentR0 which are computed in both the Metal and spec gloss case with the actual value used in the shader.

Reply:
Hello,

Thank you for your review.

- As regards the specularity part:

The metalness acts as a mask.
When the metalness is high, the corresponding specularity is equal to 
the albedo. When the metalness is low, the corresponding specularity is 
(0.04, 0.04, 0.04), default value for dielectric materials.

In my calculus I assumed that the reflectance value is equal to the 
metalness definded in the ORM texture time the metallic attribute of PBR 
Metallic-Roughness model. But I may be wrong for this part.

At the end I have the same result than using specularEnvironmentR0 for the metallic workflow so I guess my calculus is good. 
However, if the specularEnvironmentR0 is equal to what I was trying to compute it's perfect.

- Regarding the roughness:

It seems that microSurface is also computed for both the metallic-roughness and spec-glossiness models. I guess it’s better to keep microSurface instead of 1.0 – roughness to avoid additional computation.

- Thus, if you agree I will change all this part by: 
gl_FragData[PREPASS_SPECULARGLOSSINESS_INDEX] = vec4(specularEnvironmentR0, microSurface) * writeGeometryInfo;

Reply:
Hello,

Don't know why, the specularEnvironnementR0 is really darker than the baseReflectivity (for PBR spec-glossiness model)
![image](https://user-images.githubusercontent.com/44207625/168772660-0b011b81-7359-45fc-becc-d5b9d3880bfa.png)
![image](https://user-images.githubusercontent.com/44207625/168772745-7d4647bb-0175-4f84-a2de-95ea577c703e.png)

Reply:
This is because in PBR, the color are in linear space and not gamma. @CraigFeldspar do you know the expected space for post processes using it ??? We should put them all consistent across materials so either calling toLinear in Standard and such or toGamma in PBR to ensure we have the expected output.

Reply:
AFAIK, SSR is the only one using reflectivity - and it should use linear space. Calling `toLinear` in standard is the way to go.
Also, not altering the PBR + SSR output seems the best call to me

Reply:
> This is because in PBR, the color are in linear space and not gamma. @CraigFeldspar do you know the expected space for post processes using it ??? We should put them all consistent across materials so either calling toLinear in Standard and such or toGamma in PBR to ensure we have the expected output.

Both images come from PBR. But the first one calls baseReflectivity and the other specularEnvironmentR0.
So, is baseReflectivity in gamma space and specularEnvironmentR0 in linear space ?

Reply:
> AFAIK, SSR is the only one using reflectivity - and it should use linear space. Calling `toLinear` in standard is the way to go. Also, not altering the PBR + SSR output seems the best call to me

Not sure I did it well. To me specularity is currently not consistent across PBR spec-glossiness and PBR metal-roughness since I don't use specularEnvironmentR0 for both cases.

If I call toLinearSpace in default it will affect standard + SSR (reflectivity will be darker)
If I call specularEnvironmentR0 for both PBR spec-gloss and PBR metal-roughness it will alter PBR + SSR too

Reply:
It also has to be consistent with geometryBuffer data I guess ?

Reply:
As far as I understand, reflectivity was returned in gamma space everywhere (Standard, PBR spec-Glossy for both prePass and geometryBuffer).
It seems more logical to only apply toGammaSpace(specularEnvironmentR0) in order to return reflectivity of Metal-roughness material in the same space than other materials (rather than changing the space of reflectivity for Standard and PBR spec-Glossy and having to deal with regression)

Reply:
> As far as I understand, reflectivity was returned in gamma space everywhere (Standard, PBR spec-Glossy for both prePass and geometryBuffer).
> It seems more logical to only apply toGammaSpace(specularEnvironmentR0) in order to return reflectivity of Metal-roughness material in the same space than other materials (rather than changing the space of reflectivity for Standard and PBR spec-Glossy and having to deal with regression)

This makes perfect sense to me ? I guess it is the latest required change with the Geometry Buffer one that needs to be in sync.

@Mannns you rock !!!

Reply:
Hi @sebavan 

I just added changes to extend reflectivity pass of the Geometry Buffer to all material types.
Since I missed a lot of information (and in particular specularEnvironmentR0) in the G-Buffer, I computed specularity-glossiness with the formula you didn’t approve at the beginning…  The reflectivity seems to match with the one in the PrePass, but if you know an other way to compute it please tell me. 

Plus, my code in the G-Buffer doesn’t take transparency into account, but it was already not taken into account before.